Update to exception list documentation in Polaris (#7183)
Shopify/polaris: polaris.shopify.com/content/components/exception-list/index.md

- Have a description (a title is optional)
- Be concise

> Punctuation for exception lists should follow [list item guidelines](https://polaris.shopify.com/components/list). Standalone exceptions (a single sentence or sentence fragment) do not need end punctuation. Titles are optional and can be used as a header or as an inline label introducing the description.

Reviewer: ```suggestion
Punctuation for exception lists should follow [list item guidelines](https://polaris.shopify.com/components/list). Standalone exceptions (a single sentence or sentence fragment) don’t need end punctuation. Titles are optional and can be used as a header or as an inline label introducing the description.
```
